helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/WagedRebalancer.java

```diff
@@ -139,6 +141,11 @@ protected WagedRebalancer(AssignmentMetadataStore assignmentMetadataStore,
 
     // Calculate the target assignment based on the current cluster status.
     Map<String, IdealState> newIdealStates = computeBestPossibleStates(clusterData, resourceMap);
+    // Apply user-defined preference list after the calculation.
+    // Note the user-defined list will be applied as an overwritten to the final result, instead of
+    // part of the rebalance algorithm.
+    newIdealStates.entrySet().stream().forEach(idealStateEntry -> applyUserDefinePreferenceList(
```

Comment:
+1 curious as well

Reply:
In the calculation, the assignment will be persisted into the AssignmentMetadata store. This will impact the calculation. If we want to treat the user-defined list as a temporary config, we shall not persist it.
It is debatable. So let me what's your thought here.

Reply:
Talked with Junkai offline, we agreed to keep the current design.